pkg/util/util.go

```diff
@@ -281,3 +287,101 @@ func ParseInt32P(v string) (*int32, error) {
 	}
 	return types.Int32P(int32(vi)), nil
 }
+
+type WorkloadClients struct {
+	KubeClient       kubernetes.Interface
+	OcClient         oc_cs.Interface
+	StashClient      cs.Interface
+	CRDClient        crd_cs.ApiextensionsV1beta1Interface
+	AppCatalogClient appcatalog_cs.Interface
+}
+
+func (wc *WorkloadClients) IsTargetExist(target api_v1beta1.TargetRef, namespace string) bool {
+
+	fmt.Println(target.APIVersion, target.Kind, target.Name, namespace)
+	switch target.Kind {
+	case apis.KindDeployment:
+		switch target.APIVersion {
+		case apis.APIVersionAppsV1:
+			if _, err := wc.KubeClient.AppsV1().Deployments(namespace).Get(target.Name, metav1.GetOptions{}); err == nil {
```

Comment:
You don't need to check both. Just use apps/v1 regardless of what user says for Official workloads.

Reply:
removed